### PR TITLE
don't block installation on unsupported platforms

### LINKED ISSUE #146: Android support
Hello,

When trying to install dugite on Android, I get this error:
```
yarn add v1.3.2
info No lockfile found.
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
[4/4] Building fresh packages...
[1/1] ⠠ dugite
[-/1] ⠐ waiting...
[-/1] ⠐ waiting...
[-/1] ⠐ waiting...
error /data/data/com.termux/files/home/node_modules/dugite: Command failed.
Exit code: 1
Command: node ./script/download-git.js
Arguments: 
Directory: /data/data/com.termux/files/home/node_modules/dugite
Output:
/data/data/com.termux/files/home/node_modules/dugite/script/config.js:32
  const index = pathName.lastIndexOf('/')
                         ^

TypeError: Cannot read property 'lastIndexOf' of null
    at getConfig (/data/data/com.termux/files/home/node_modules/dugite/script/config.js:32:26)
    at Object.<anonymous> (/data/data/com.termux/files/home/node_modules/dugite/script/download-git.js:11:35)
    at Module._compile (module.js:635:30)
    at Object.Module._extensions..js (module.js:646:10)
    at Module.load (module.js:554:32)
    at tryModuleLoad (module.js:497:12)
    at Function.Module._load (module.js:489:3)
```

This is because `process.platform == 'android'`.

I don't expect you to provide an ARM Android binary but I understand we can use another Git binary with `GIT_EXEC_PATH`, so I guess the install script could just ignore the download (maybe with a warning) when on Android.

### PR DESCRIPTION
**Pull request by shiftkey**
Fixes #146:

- [x] only download if a source is defined
- [x] for other platforms, point over to resource about environment variables
- [x] finish documentation about environment variables

### CODE REVIEW

**Comment by Rudloff** [comment]
Additionally, you could add something like this to `package.json` so that NPM/Yarn can throw an error early in the install process:
```json
  "os": [
    "darwin",
    "linux",
    "win32"
  ]
```

**Comment by shiftkey** [comment]
@Rudloff that's a neat idea, but as we support invoking an external Git we don't necessarily want to remove support completely

**Comment by shiftkey** [comment]
@Rudloff how does this feel?

File: script/download-git.js [hunk 1]
```diff
@@ -99,7 +99,15 @@ const downloadAndUnpack = () => {
   })
 }
 
-mkdirp(config.outputPath, function (error) {
+if (config.source === '') {
+  console.log(
+    `Skipping downloading embedded Git as platform '${process.platform}' is not supported.`
+  )
+  console.log(`To learn more about using dugite with a system Git: https://git.io/vF5oj`)
```
Comment (shiftkey) [inline script/download-git.js]: This shortened URL will unwrap to `https://github.com/desktop/dugite/blob/master/docs/environment-variables.md#execution` which will work once the PR is merged.

**Comment by Rudloff** [comment]
Sorry, by quickly looking at your PR, I thought you only wanted to display a nicer error on unsupported platforms.
If you skip the download and allow using a custom Git binary, that's even better :smiley:

**Comment by shiftkey** [comment]
> If you skip the download and allow using a custom Git binary, that's even better 😃

Yeah, there's no good reason for me to prevent you from using it if you know what you're doing. And if I get feedback that it's the wrong thing, I'll revisit it.

**Pull request merged**